src/metabase/query_processor/util.clj

```diff
@@ -94,6 +94,7 @@
   (let [{:keys [constraints parameters], :as query} (select-keys query [:database :lib/type :stages :parameters :constraints])]
     (cond-> query
       (empty? constraints) (dissoc :constraints)
+      true                 (update :parameters  (fn [params] (mapv #(if (sequential? (:value %)) (update % :value sort) %) params)))
```

Comment:
nit - this might be enough logic to be worth pulling out to a helper function, like `walk-query-sort-maps`